### DISCUSSION TITLE
Search Query with multiple values in taxonomy user property

### DISCUSSION THREAD

**Discussion by FloWaeger** [microsoft-search/pnp-modern-search#1976] [Q&A]
We have a custom user property ("UserAbo") which is mapped to a TermSet. I want to filter all contents (SiteColum mapped with RefinableString100) with at least one of my picked terms within my user profile.

So my query is {|RefinableString100:{User.UserAbo}} but I think the query is not resolved correctly.

![image](https://user-images.githubusercontent.com/102596638/160608998-bbcb6246-c9d3-4339-9941-59c0659d2e8a.png)

Is my query not correct?

Best regards
Florian

**Comment by wobba** [comment]
If `UserAbo` is the internal name of your UPA and it's available to read then it should be replaced. I recommend looking at the JSON request and see if it's replaced to blank before being sent or if the query variable is sent over. It also helps to know what version of the web part you are using, meaning have you tried with the latest release.

**Reply by FloWaeger** [reply]
UserAbo is the internal name
Using Version 4.5.4

The variable is replaced but the query should be (RefinableString100:Term1) OR (RefinableString100:Term2), right?
Query looks equal within network tab

![image](https://user-images.githubusercontent.com/102596638/160611116-660f48e4-8f2e-4491-9113-2d38729ac902.png)

**Reply by FranckyC** [reply]
In the current implementation, the operator is expecting values separated by a comma ',' but it appears they are using `|` instead. Can you try to create your property with the following settings:

![image](https://user-images.githubusercontent.com/5659122/160640996-70746920-f919-441b-b270-277c50d7ef47.png)

**Reply by FloWaeger** [reply]
Property settings are following:
![image](https://user-images.githubusercontent.com/102596638/160642740-3432b99b-fee7-40f4-a2a5-de80c77942e8.png)

Even the SPS-Interests property is set up like you mentioned and delivers the same result within the query:
![image](https://user-images.githubusercontent.com/102596638/160642694-f8a73b6b-4b9e-4dbf-821f-f28df6b4b1ae.png)

**Reply by wobba** [reply]
This was fixed in https://github.com/microsoft-search/pnp-modern-search/pull/1879 so will be correct in the next release.
